app/lib/efile/id/id40_calculator.rb

```diff
@@ -207,8 +210,29 @@ def calculate_line_46
           @intake.state_file1099_rs.sum { |item| item.state_tax_withheld_amount.round }
       end
 
-      def calculate_line_54; end
-      def calculate_line_56; end
+      def calculate_line_50
+        line_or_zero(:ID40_LINE_43) + line_or_zero(:ID40_LINE_46)
+      end
+
+      def calculate_line_51
+        if line_or_zero(:ID40_LINE_42) > line_or_zero(:ID40_LINE_50)
+          line_or_zero(:ID40_LINE_42) - line_or_zero(:ID40_LINE_50)
+        end
```

Comment:
should the `else` be specified here as `0` (in the case line 50 > line 40 value) or is that not a significant difference at the end of the day since `line_or_zero` is being called on the `:ID40_LINE_51` value?

Reply:
@arinchoi03 these elements would not appear if they are equal values which is the intention. I'll double check with Margaret but I don't think we want to display a refund or payment if they don't have one (is zero). And since the refund_or_owed_amount method in the base calculator class also uses `line_or_zero` I don't think we'll run into issues there either. Sorry did that cover your concerns or did I miss something?